src/Illuminate/Support/Collection.php

```diff
@@ -1548,9 +1548,9 @@ public function offsetSet($key, $value)
     {
         if (is_null($key)) {
             $this->items[] = $value;
-        } else {
-            $this->items[$key] = $value;
         }
+
+        $this->items[$key] = $value;
```

Comment:
@geekcom: you could've only done that shortening if you were breaking the path of execution, using [guard clauses](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) for instance. I wouldn't recommend that though, since it alters type signature.